### PR TITLE
 Pointed adjustments and new features

### PR DESCRIPTION
**Pull request by renandvp**
- Rules to enter the game

- Ranking page

- Adding points when guessing the word

- Added color palette

### CODE REVIEW

File: public/js/draw.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
 document.addEventListener("DOMContentLoaded", function () {
     const CANVAS_UPDATE_RATE = 25; //Miliseconds
-    const width = window.innerWidth;
-    const height = window.innerHeight;
+    const width = 1400;
```
Comment (jeansouzak) [inline public/js/draw.js]: Why static?
Will not work in other screen resolution

File: public/js/draw.js [hunk 2]
```diff
@@ -20,6 +20,25 @@ document.addEventListener("DOMContentLoaded", function () {
     setupSocket();
     startLoop();
 
+    var colors = document.getElementsByClassName('color');
+    for (var i = 0; i < colors.length; i++){
+        colors[i].addEventListener('click', onColorUpdate, false);
```
Comment (jeansouzak) [inline public/js/draw.js]: You are using JQuery on whole code.
Why do you use this way here?

File: public/js/draw.js [hunk 3]
```diff
@@ -20,6 +20,25 @@ document.addEventListener("DOMContentLoaded", function () {
...
+    }
+    
+    function onColorUpdate(e){
+        drawingData.color = e.target.className.split(' ')[1];
```
Comment (jeansouzak) [inline public/js/draw.js]: You are using JQuery on whole code.
Why do you use this way here?

File: public/index.html [hunk 4]
```diff
@@ -1,19 +1,24 @@
...
+	<body>
+		<div class="login-form">
+			<h1>Shared Draw Game</h1>
+			<form action="draw" method="POST">
+				<input type="text" name="name" id="name" placeholder="name" required>
+				<input type="text" name="word" id="word" placeholder="word" required>
+				<input type="submit" id="register">
+			</form>
+		</div>
+		<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
```
Comment (jeansouzak) [inline public/index.html]: JQuery 2.1?
image

File: server.js [hunk 5]
```diff
@@ -2,23 +2,56 @@ const express = require('express');
...
+
+app.post('/draw', (req, res) => {
+    let newUser = new user(req.body.name, req.body.word);
+    let player = service.findPlayer(newUser.name, players);
+    if(!player){
+      players.push(newUser);
+      res.render('draw', newUser);
+    } else {
+    //TODO: error message
+    res.redirect('/');
```
Comment (jeansouzak) [inline server.js]: Ident?

File: server.js [hunk 6]
```diff
@@ -2,23 +2,56 @@ const express = require('express');
...
+  }
+})
+
+app.post('/send-word', (req, res) => {
+    if(service.hasWord(req.body.word, players)){
```
Comment (jeansouzak) [inline server.js]: Player list could be there in service / repository package

File: server.js [hunk 7]
```diff
@@ -2,23 +2,56 @@ const express = require('express');
...
+      winnerPlayer = service.findPlayer(req.body.name, players);
+      //TODO: Rules to not guess your own word
```
Comment (jeansouzak) [inline server.js]: thats important haha

File: server.js [hunk 8]
```diff
@@ -2,23 +2,56 @@ const express = require('express');
...
+      if(winnerPlayer){
+        winnerPlayer.points++;
+      }
+      res.send(winnerPlayer);
+    }
+});
 
-//History with drawed lines to render on new client connection
-const drawedList = [];
```
Comment (jeansouzak) [inline server.js:12]: why delete drawedList?
Other players or refresh page should display draws on screen

File: public/views/ranking.html [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+    <% players.forEach(function(user) { %>
+        <tr>
+            <td><%= user.name %></td>
+            <td><%= user.word %></td>
+            <td><%= user.points %></td>
+        </tr>
+      <% }); %>
+</table>
+    <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
```
Comment (jeansouzak) [inline public/views/ranking.html]: JQuery 2.1?

File: server.js [hunk 10]
```diff
@@ -2,23 +2,56 @@ const express = require('express');
...
+app.post('/ranking', (req, res) =>{
+  players.sort(service.dynamicSort("points","desc"));
```
Comment (jeansouzak) [inline server.js]: It is a generic service?
Like a tools

Comment (jeansouzak) [inline server.js]: You could sort simple:
```
players.sort((a, b) => b.points - a.points);
```

File: server.js [hunk 11]
```diff
@@ -2,23 +2,56 @@ const express = require('express');
...
+})
+
+app.post('/send-word', (req, res) => {
+    if(service.hasWord(req.body.word, players)){
+      winnerPlayer = service.findPlayer(req.body.name, players);
+      //TODO: Rules to not guess your own word
+      if(winnerPlayer){
+        winnerPlayer.points++;
+      }
+      res.send(winnerPlayer);
```
Comment (jeansouzak) [inline server.js]: Why named winnerPlayer?

File: app/services.js [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+        var i;
+        for (i = 0; i < players.length; i++) {
+          if (players[i].name === name) {
+            return players[i];
+          }
+        }
+        return false;
+    },
+
+    dynamicSort: function(property,order) {
```
Comment (jeansouzak) [inline app/services.js]: o.o'

File: app/services.js [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+        for (i = 0; i < players.length; i++) {
+          if (players[i].word === word) {
+            return true;
+          }
+        }
+        return false;
+    },
+
+    findPlayer: function (name, players) {
+        var i;
```
Comment (jeansouzak) [inline app/services.js]: You could use:
```
return players.some(player => player.name === name)
```

File: app/services.js [hunk 14]
```diff
@@ -0,0 +1,38 @@
+var self = module.exports = {
+    hasWord: function (word, players) {
+        var i;
+        for (i = 0; i < players.length; i++) {
```
Comment (jeansouzak) [inline app/services.js]: return players.some(player => player.word === word)

File: app/tools.js [hunk 15]
```diff
@@ -0,0 +1,15 @@
+var self = module.exports = {
+    hasWord: function (word, players) {
+      return players.some(player => player.word === word)
+    },
+
+    findPlayer: function (name, players) {
+      var i;
+      for (i = 0; i < players.length; i++) {
+        if (players[i].name === name) {
+          return players[i];
```
Comment (renandvp) [inline app/tools.js]: I need the returned payer here

File: server.js (hunk 6 above)
Comment (renandvp) [reply server.js]: Gonna fix that yet

File: server.js (hunk 7 above)
Comment (renandvp) [reply server.js]: Next feature, I promisse

File: server.js (hunk 8 above)
Comment (renandvp) [reply server.js:12]: I put it back

File: app/tools.js (hunk 15 above)
Comment (jeansouzak) [reply app/tools.js]: https://lmgtfy.com/?q=javascript+find+object+in+array

File: server.js [hunk 16]
```diff
@@ -2,23 +2,64 @@ const express = require('express');
...
+})
+
+app.post('/send-word', (req, res) => {
+    let thisName = req.body.name;
+    let thisWord =  req.body.word;
+    if(thisName && thisWord){
+      io.emit('chat', thisName, thisWord);
+    }
+    if(tools.hasWord(thisWord, players)){
+      thisUser = tools.findPlayer(thisName, players);
```
Comment (jeansouzak) [inline server.js:39]: service / repository to find player

**Pull request merged**